### PR TITLE
8292376: A few Swing methods use inheritDoc on exceptions which are not inherited

### PR DESCRIPTION
**Pull request by kumarabhi006**
Document update for getBaselineResizeBehavior() and getBaseline() method description in JSpinner, AbstractBorder and TitledBorder .

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change requires a CSR request to be approved

### Issues
 * [JDK-8292376](https://bugs.openjdk.org/browse/JDK-8292376): A few Swing methods use inheritDoc on exceptions which are not inherited
 * [JDK-8293470](https://bugs.openjdk.org/browse/JDK-8293470): A few Swing methods use inheritDoc on exceptions which are not inherited (**CSR**)

### Reviewers
 * [Phil Race](https://openjdk.org/census#prr) (@prrace - **Reviewer**)
 * [Tejesh R](https://openjdk.org/census#tr) (@TejeshR13 - Author)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/9902/head:pull/9902` \
`$ git checkout pull/9902`

Update a local copy of the PR: \
`$ git checkout pull/9902` \
`$ git pull https://git.openjdk.org/jdk pull/9902/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 9902`

View PR using the GUI difftool: \
`$ git pr show -t 9902`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/9902.diff">https://git.openjdk.org/jdk/pull/9902.diff</a>

</details>

### CODE REVIEW

**Comment by kumarabhi006** [comment]
1) In javax.swing.JSpinner.DefaultEditor.getBaselineResizeBehavior() method, the exception tag has been removed. Neither this method nor any of the ancestor class definition of this method declared exception tag in description.

2) For javax.swing.border.TitledBorder.getBaseline(java.awt.Component,int,int) method, the ancestor class definition of this method didn't declare exception tag for NPE. So, {@inheritDoc} didn't inherit anything. 
However, javax.swing.border.TitledBorder.getBaseline(java.awt.Component,int,int) throws NPE if Component is null. NPE tag has been changed accordingly.

3) For javax.swing.border.TitledBorder.getBaselineResizeBehavior(java.awt.Component) method, the {@inheritDoc} didn't inherit anything because there was no exception tag in ancestor class method definition.
As TitledBorder extends AbstractBorder, getBaselineResizeBehavior(java.awt.Component) method throws an exception if Component is null but the exception tag was missing for method description in AbstractBorder.
So, exception tag for NPE added for getBaselineResizeBehavior(java.awt.Component) method in AbstractBorder.
Now in  javax.swing.border.TitledBorder.getBaselineResizeBehavior(java.awt.Component) method, the {@inheritDoc} inherits NPE tag from AbstractBorder.

File: src/java.desktop/share/classes/javax/swing/border/AbstractBorder.java [hunk 1]
```diff
@@ -181,6 +181,7 @@ public int getBaseline(Component c, int width, int height) {
      * @param c <code>Component</code> to return baseline resize behavior for
      * @return an enum indicating how the baseline changes as the border is
      *         resized
+     * @throws NullPointerException if passed in {@code Component} is {@code null}
```
Comment (TejeshR13) [inline src/java.desktop/share/classes/javax/swing/border/AbstractBorder.java:184]: `if passed in {@code Component} is {@code null}` - Rephrasing this sentence would be better. Something like `if {@code Component} is {@code null}`.

Comment (kumarabhi006) [reply src/java.desktop/share/classes/javax/swing/border/AbstractBorder.java]: Updated in method description.

**Comment by kumarabhi006** [comment]
Attached the CSR link
https://bugs.openjdk.org/browse/JDK-8293470

**Comment by prrace** [comment]
The bug report has no evaluation.
The PR just says "Doc update for Null Pointer Exception"
The bug should always have a proper evaluation and the PR can then refer to that or copy it.
The CSR has a lot of what should have appeared in these places too ..

Here's my take on it

The bug complains of 3 methods incorrectly using inheritdoc for NPE
and the resolution for each case is as follows

- javax.swing.JSpinner.DefaultEditor.getBaselineResizeBehavior()
It is unclear why the inheritdoc appears on this method.
This nested class extends JPanel and so the method is inherited from JComponent
and then in turn from Component and neither of these seems like it will throw NPE
The fix is to remove it - doc only no behavioural change.

- javax.swing.border.TitledBorder.getBaseline(java.awt.Component,int,int)
The superclass (AbstractBorder) method cannot throw NPE so inheritdoc is wrong.
However this method can throw NPE so the fix is to replace it with appropriate
documentation for this method. So also doc only no behavioural change.

- javax.swing.border.TitledBorder.getBaselineResizeBehavior(java.awt.Component)
 This method delegates to the super class method which can throw NPE but omits to specify it.
 The fix is to add the declaration to the super class method and then the inheritdoc
 is appropriate and can be left in place.  Again doc only no behavioural change.

So the changes look correct - it just is better with an explanation.

**Review by prrace — APPROVED** [review]

**Review by TejeshR13 — APPROVED** [review]

**Pull request closed without merging**